Mathlib/CategoryTheory/RespectsIso.lean

```diff
@@ -0,0 +1,60 @@
+/-
+Copyright (c) 2024 Joël Riou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joël Riou
+-/
+import Mathlib.CategoryTheory.Iso
+import Mathlib.Data.Set.Basic
+
+/-! # Sets of objects which respects isomorphisms
+
+This file introduces the type class `S.RespectsIso` for `S : Set C`
+with `C` is category.
+
+-/
+
+open CategoryTheory
+
+variable {C : Type*} [Category C]
+
+namespace Set
+
+variable (S T : Set C)
+
+/-- A set `S : Set C` respects isomorphisms if whenever `X ∈ S`, then all the
+objects that are isomorphic to `X` also belong to `S`.  -/
+class RespectsIso : Prop where
+  condition {X Y : C} (_ : X ≅ Y) (_ : X ∈ S) : Y ∈ S
+
+lemma mem_of_iso [S.RespectsIso] {X Y : C} (e : X ≅ Y) (hX : X ∈ S) : Y ∈ S :=
```

Comment:
Since there is no data here, I think it makes sense to phrase `mem_of_iso` and `mem_iff_of_iso` in terms of `IsoIso` rather than `Iso` (or maybe in addition to the existing results).

Reply:
I have added the `IsIso` variants. (In the applications, I use more `Iso` than `IsIso`.)